src/utils/discord.ts

```diff
@@ -1,110 +1,133 @@
-  const server = bot.guilds.cache.get(env.discord.server);
-  server.roles.create({
-    data: {
-      name: teamName,
-      color: 'FFEBAB',
-    },
+  return createDiscordChannel({
+    name: channelName,
+    type: channelType,
+    parent_id: tournament.discordCategoryId,
+    permission_overwrites: [
+      {
+        id: everyoneRole.id,
+        type: DiscordChannelPermissionType.ROLE,
+        allow: DiscordChannelPermission.DEFAULT,
+        deny: DiscordChannelPermission.VIEW_CHANNEL,
+      },
+      {
+        id: tournament.discordRespoRoleId,
+        type: DiscordChannelPermissionType.ROLE,
+        allow: DiscordChannelPermission.VIEW_CHANNEL,
+        deny: DiscordChannelPermission.DEFAULT,
+      },
+      {
+        id: teamRole.id,
+        type: DiscordChannelPermissionType.ROLE,
+        allow: DiscordChannelPermission.VIEW_CHANNEL,
+        deny: DiscordChannelPermission.DEFAULT,
+      },
+    ],
   });
 };
 
-// eslint-disable-next-line arrow-body-style
-export const setupTeam = async (team: string, tournamentId: TournamentId) => {
-  const tournament = await fetchTournament(tournamentId);
+export const setupDiscordTeam = async (team: Team, tournament: Tournament) => {
+  if (!env.discord.token) {
+    logger.warn('Discord token missing. It will skip discord calls');
+    return;
+  }
+
   // Only create channel if there are more than 1 player per team
   if (tournament.playersPerTeam !== 1) {
-    return Promise.all([
-      createRole(team),
-      createChannel(team, 'text', tournamentId),
-      createChannel(team, 'voice', tournamentId),
+    const role = await createDiscordRole({ name: team.name, color: env.discord.teamRoleColor });
+
+    logger.debug(`Create discord channels for ${team.name}`);
+    // Create the channels and update in the database the role.
+    await Promise.all([
+      createDiscordTeamChannel(team.name, DiscordChannelType.GUILD_TEXT, tournament.id, role),
+      createDiscordTeamChannel(team.name, DiscordChannelType.GUILD_VOICE, tournament.id, role),
+      database.team.update({ data: { discordRoleId: role.id }, where: { id: team.id } }),
     ]);
   }
-  return null;
 };
 
 export const syncRoles = async () => {
-  // Get the server, loop all tournaments, loop all teams in tournaments and loop all player in teams to give them the role
-  const server = bot.guilds.cache.get(env.discord.server);
+  if (!env.discord.token) {
+    logger.warn('Discord token missing. It will skip discord calls');
+    return;
+  }
 
+  // Get the server, loop all tournaments, loop all teams in tournaments and loop all player in teams to give them the role
   for (const tournament of await fetchTournaments()) {
     // Do not care about the solo tournaments
     if (tournament.playersPerTeam === 1) {
       continue;
     }
-    const tournamentRole = server.roles.cache.find((role) => role.id === tournament.discordRoleId);
+    const tournamentRole = await findDiscordRoleById(tournament.discordRoleId);
 
```

Comment:
Je pensais à cette boucle for pour le `fetchDiscordRoles`
```ts
  const discordRoles = await fetchDiscordRoles();
  // Get the server, loop all tournaments, loop all teams in tournaments and loop all player in teams to give them the role
  for (const tournament of await fetchTournaments()) {
    // Do not care about the solo tournaments
    if (tournament.playersPerTeam === 1) {
      continue;
    }
    const tournamentRole = discordRoles.find((role) => role.id === tournament.discordRoleId);
    ...
  }
```
Edit: à quoi ça sert de récupérer le rôle en le filtrant par id si ce n'est que pour utiliser son id après ? Si c'est pour vérifier qu'il existe, il faut le créer s'il n'existe pas.